### ISSUE TITLE
Multi-cursor not indenting block of code as should

### ISSUE THREAD

**Issue by Strapchay** [zed-industries/zed#10500]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Using the multiple line cursor to indent a block of code doesn't function as should in relation to vscode. Instead of treating each line of the code as a line, for a single line with some formatting, it chips into those content and indents them as well. I enter the multiple line cursor mode using Alt + shift.
![before_indent_mail](https://github.com/zed-industries/zed/assets/156309032/6943ec34-0545-4b76-8d60-2abf1f7937ca)
![before_indent](https://github.com/zed-industries/zed/assets/156309032/6f0e0dc4-a3f4-4daa-ae92-8de96724e8cb)
![after_indent_mail](https://github.com/zed-industries/zed/assets/156309032/b0b8eefa-0141-4cfc-a41f-0ead276fb943)
![after_indent](https://github.com/zed-industries/zed/assets/156309032/24013db8-e5fd-4caf-8e36-a357f7bb11d9)

### Environment

Zed Dev: v0.132.0 (Zed)
OS: Linux
Memory: 16GiB
Architecture: x86_64

**Comment by Strapchay** [comment]
Yeah, i am still able to reproduce it, it doesn't indent code blocks rightly most especially after the lines to indent is selected and the tab button is pressed to push them rightward, the indentation collapses to a straightline and disregards the indentation

**Comment by beniaminzagan** [comment]
@Strapchay do you have soft wrap on? Can we get the exact same text as above to try and reproduce? 
Cause If I have word wrap on I can reproduce something similar. But not if its off. I tried in Zed Preview 175.1. Also, vscode with the same setup does the same thing.

**Comment by Strapchay** [comment]
Let me try with word wrap off and see if it's reproducible then, I usually have word wrap on most of the time. However, the specific case as in the example i think is after the code was formatted. Would let you knoe how that goes

**Comment by lionel-** [comment]
@SomeoneToIgnore This is still an issue that makes the combination of virtually wrapped line and multiple cursors very hard to use. I don't think I've ever used another editor that behaves like this. Could you please comment more on why fixing this is not planned?

**Comment by SomeoneToIgnore** [comment]
Right above your message, in the previous one, people asked for the reproduction, in _February_, and got no answer still — that is the reason for closing it as nothing actionable.

**Comment by lionel-** [comment]
@SomeoneToIgnore Gotcha! Would you mind reopening https://github.com/zed-industries/zed/issues/25237 that was closed as a duplicate of this issue? I'm available for follow up questions.
